### PR TITLE
feat: Add `<ProductBadge />` support to `<Card />` component

### PR DESCRIPTION
**Pull request by nandereck**
🎟️ [Asana Task](https://app.asana.com/0/1202802428287999/1203049731246701/f)
🔍 [Preview Link](https://react-components-git-nels-cardadd-product-badg-3c55a7-hashicorp.vercel.app/components/card)

---

## Description

This PR enables `<ProductBadge />` support to the `<Card />` component. Couple of call-outs:
- The spec for the `Card` component displays the `hasDot` variant of `ProductBadge`. Given that and a preference to be a bit restrictive, we force that variant in the code.
- Similarly, the code does not allow `ProductBadge` to be a composable child.

## Validation Steps
- Navigate to the [preview link](https://react-components-git-nels-cardadd-product-badg-3c55a7-hashicorp.vercel.app/components/card)
- Add the following prop to any of the cards in the Code Editor:
`productBadges={{badges: [{ productName: 'waypoint'}]}}`
- Verify that the Product Badges render properly
- Review the Use Case Cards in the Examples section

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/card/index.tsx [hunk 1]
```diff
@@ -33,6 +37,14 @@ function Card(props: CardProps) {
...
             {meta && meta.length > 0 ? <Meta items={meta} /> : null}
             <Heading>{heading}</Heading>
+            {productBadges &&
+            productBadges?.badges &&
+            productBadges?.badges?.length > 0 ? (
+              <ProductBadges
+                badges={productBadges.badges}
+                appearance={appearance}
+              />
+            ) : null}
```
Comment (alexcarpenter) [inline packages/card/index.tsx:47]: Looks like badges should display below the description.

CleanShot 2022-10-05 at 17 01 24@2x

File: packages/card/index.tsx [hunk 2]
```diff
@@ -101,6 +113,25 @@ function Heading({ as: Component = 'h2', children }: HeadingProps) {
   )
 }
 
+function ProductBadges({ badges, appearance = 'light' }: ProductBadgesProps) {
```
Comment (alexcarpenter) [inline packages/card/index.tsx:116]: Cards can only render productBadges with the dot and secondary theme?

File: package.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
     "@hashicorp/flight-icons": "^2.5.0",
     "@hashicorp/mktg-global-styles": "^4.2.0",
     "@hashicorp/mktg-logos": "^1.3.2",
+    "@hashicorp/react-product-badge": "^0.1.1",
```
Comment (dstaley) [inline package.json:8]: I don't think we want this dependency installed at the root. You can fix this by running the following in the root directory:

```
npm uninstall @hashicorp/react-product-badge
npm install @hashicorp/react-product-badge --workspace=@hashicorp/react-card
```

That will ensure that consumers who install `@hashicorp/react-card` get the `@hashicorp/react-product-badge` dependency.

File: packages/card/index.tsx (hunk 1 above)
Comment (nandereck) [reply packages/card/index.tsx:48]: Oops - you're right. [Moved](https://github.com/hashicorp/react-components/pull/748/commits/fbb86263b5577426c921f2bc54d978c61a3afd48)!

File: packages/card/index.tsx [hunk 4]
```diff
@@ -101,6 +113,25 @@ function Heading({ as: Component = 'h2', children }: HeadingProps) {
...
+  return (
+    <div className={s.productBadges}>
+      {badges.map((badge, stableIdx) => {
+        return (
+          // eslint-disable-next-line react/no-array-index-key
+          <React.Fragment key={stableIdx}>
```
Comment (pbortnick) [inline packages/card/index.tsx:122]: ❓ Do we need `<React.Fragment>` here or can we add the key directly to `<ProductBadge>` ?

**Review by dstaley — CHANGES_REQUESTED** [review]

File: packages/card/index.tsx (hunk 2 above)
Comment (nandereck) [reply packages/card/index.tsx:116]: See my comment above in the PR description.

Tangentially related, but `hasDot` is the `secondary` theme plus a dot versus having totally separate styles if that makes sense.

File: packages/card/index.tsx (hunk 4 above)
Comment (pbortnick) [reply packages/card/index.tsx:122]: If not, can we also delete the React import?

File: packages/card/index.tsx (hunk 2 above)
Comment (alexcarpenter) [reply packages/card/index.tsx:116]: Got it, sorry I missed that in the description!

Thoughts on simplifying the prop if we're being restrictive here.

```tsx
<Card
  productBadges={['vault', 'consul']}
/>
```

File: packages/card/index.tsx (hunk 4 above)
Comment (nandereck) [reply packages/card/index.tsx]: Oops - good catch. This is cruft. [Removed](https://github.com/hashicorp/react-components/pull/748/commits/3be3b6a09666b959e6e73906a7bb10533fd0bca1)!

File: package.json (hunk 3 above)
Comment (nandereck) [reply package.json:8]: Gah, my bad! Will do!

Comment (nandereck) [reply package.json]: Commit [here](https://github.com/hashicorp/react-components/pull/748/commits/0d5334ef7c6a10c3f1cc292b992dd1e8f64b2b56).

**Review by dstaley — APPROVED** [review]

File: packages/card/index.tsx [hunk 5]
```diff
@@ -34,6 +37,14 @@ function Card(props: CardProps) {
             {meta && meta.length > 0 ? <Meta items={meta} /> : null}
             <Heading>{heading}</Heading>
             {description ? <Description>{description}</Description> : null}
+            {productBadges &&
+            productBadges?.badges &&
+            productBadges?.badges?.length > 0 ? (
+              <ProductBadges
+                badges={productBadges.badges}
+                appearance={appearance}
```
Comment (alexcarpenter) [inline packages/card/index.tsx:45]: Maybe this is expected, but noticed the badge color is the same as the cards hover background color when appearance is dark.

CleanShot 2022-10-05 at 17 23 11@2x

Comment (nandereck) [reply packages/card/index.tsx:45]: Let me double check, but I'm not sure this is accounted for in either the `Card` or `ProductBadge` spec. 😕

File: packages/card/types.ts [hunk 6]
```diff
@@ -38,6 +41,11 @@ export interface HeadingProps {
   children: string
 }
 
+export interface ProductBadgesProps {
+  badges: Array<ProductBadgeProps['productName']>
```
Comment (nandereck) [inline packages/card/types.ts:45]: @alexcarpenter - thoughts on this? I've actually never utilized lookup types. 🙃

**Review by alexcarpenter — APPROVED** [review]
LGTM!

**Pull request merged**